src/modules/deal-folder/deal-folder.service.ts

```diff
@@ -0,0 +1,191 @@
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { CASE_LIBRARY, DOCUMENT_X0020_STATUS, ENUMS, MAX_FILE_SIZE_BYTES } from '@ukef/constants';
+import { DocumentTypeEnum } from '@ukef/constants/enums/document-type';
+import { SharepointResourceTypeEnum } from '@ukef/constants/enums/sharepoint-resource-type';
+import { DocumentTypeMapper } from '@ukef/modules/deal-folder/document-type-mapper';
+import { DtfsStorageFileService } from '@ukef/modules/dtfs-storage/dtfs-storage-file.service';
+import GraphService from '@ukef/modules/graph/graph.service';
+
+import { UploadFileInDealFolderResponseDto } from './dto/upload-file-in-deal-folder-response.dto';
+type RequiredConfigKeys = 'baseUrl' | 'ukefSharepointName' | 'estoreDocumentTypeIdFieldName';
+
+@Injectable()
+export class DealFolderService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof SharepointConfig>, RequiredConfigKeys>,
+    private readonly documentTypeMapper: DocumentTypeMapper,
+    private readonly dtfsStorageFileService: DtfsStorageFileService,
+    private readonly graphService: GraphService,
+  ) {}
+
+  async uploadFileInDealFolder(
+    fileName: string,
+    fileLocationPath: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+    documentType: DocumentTypeEnum,
+  ): Promise<UploadFileInDealFolderResponseDto> {
+    const fileSizeInBytes = await this.checkFileIsNotTooLargeAndReturnSizeInBytes(fileName, fileLocationPath);
+
+    const file = await this.downloadFileFromDtfs(fileName, fileLocationPath);
+
+    await this.uploadFileToSharepoint(file, fileSizeInBytes, fileName, dealId, buyerName, ukefSiteId);
+
+    const encodedFullFilePath = await this.updateFileInformationInSharepoint(fileName, dealId, buyerName, ukefSiteId, documentType);
+
+    return {
+      fileUpload: encodedFullFilePath,
```

Comment:
@avaitonis The Mulesoft code returns the `FileRef` property on the list item with the information about the newly uploaded file. However, because Mulesoft uses the Sharepoint REST API (via the Sharepoint Connector), whereas the Node version of eStore uses the Microsoft Graph API, not all the properties are the same, and indeed the `FileRef` property doesn't appear to exist on Microsoft Graph's list items.

I'm currently returning the encoded (e.g., with spaces replaced by %20) form of the file path, e.g., `Buyer%20Name/D%200030000322/fileName.fileExtension`. I just wanted to check that this is the same thing as what Mulesoft returns?

Reply:
@oscar-richardson-softwire, @avaitonis is going to check what the response looks like on Mulesoft

We had a look at DTFS and noticed it doesn't use the response (other than to log it) so if we're having trouble returning the same thing that Mulesoft returns then we could suggest returning something else and check with @abhi-markan if that'd still be good for DTFS

Reply:
From Mulesoft logs I can see that old/current response is single object
{
  "fileUpload": "/sites/00703107/CaseLibrary/Exporter_name_example/D 0030123456/file_1.docx"
}